PRINCIPLES.md

```diff
@@ -0,0 +1,186 @@
+  (I would agrue that key rotation is always better option in this case, but for PII and proprietary info this isn't possible).
+- In addition, state stores must preserve information regarding the identity of agents that create new versions.
+- What Why, who when. 
+- state store should
+- Business rules and security rules about the acceptable state of the system shoudl be checked against any new proposed version of the desired configuration before such changes are accepted as a new version. (Gates before version)
+- where human process is integrated
+- changes as transactions
+- Even though we say git, it's actually only true when configured in a very particular way.
+- For reliable software operations, It's insufficient to have code define the infrastructure, it must instead be data.
+
+
+
+### 3. Continuous state reconciliation
+
+> Software agents continuously, and automatically, compare a system's _Actual State_ to its _Desired State_.
+> If the actual and desired states differ, automated actions are immediately attempted to reconcile them.
+> These differences could be due to the actual state drifting from the desired state, or the desired state changing intentionally.
+
+- If the software agents fail to bring the system's state in line with its desired state, a human operator is notified.
+- Software agents continuously check that the running system under management matches the desired state configuration, and if it does not, immediately either take remedial action to bring the system back in line with stated expectations or, if this cannot be done, alert a human operator that the system is no longer meeting expectations
+- Automated delivery: Delivery of the declarative descriptions, from the repository to runtime environment, is fully automated.
+- Software Agents: Reconcilers maintain system state and apply the resources described in the declarative configuration.
+- Closed loop: Actions are performed on divergence between the version controlled declarative configuration and the actual state of the target system.
+<!-- Please emphasize the following point to aid adoption of GitOps to risk-averse teams (i.e., almost everyone in Enterprise) -->
+- This automation should serve it's human users.
+  To this point, reconciliation automation for GitOps should include a way for humans to temporarily take back the reins as needed.
+
+### 4. Operations through declaration
+
+> When wishing to operate on a software system, a human or software agent will not interact with the running system and modify it directly.
+> Instead, the agent will create a new declarative version of the desired state in the state store.
+
+- All normal operations should occur via the creation of a new uniquely named version, not through direct interaction ith the system under management
+- Break glass - exceptions are acceptable and quite likely. and credentials to access and mutate the system must still be available in almost all cases.
```

Comment:
Let's rephrase this as not something that is acceptable and likely, but instead the goal is to move towards an elimination of all events that trigger manual interventions.